okta/okta.go

```diff
@@ -0,0 +1,242 @@
+package okta
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/evergreen-ci/gimlet"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	verifier "github.com/okta/okta-jwt-verifier-golang"
+	"github.com/pkg/errors"
+)
+
+// CreationOptions specify the options to create the manager.
+type CreationOptions struct {
+	ClientID     string
+	ClientSecret string
+	RedirectURI  string
+	Issuer       string
+}
+
+type oktaUserManager struct {
+	clientID     string
+	clientSecret string
+	redirectURI  string
+	issuer       string
+
+	// TODO (kim): token caching functions
+}
+
+// NewUserManager creates a manager that connects to Okta for user
+// management services.
+func NewUserManager(opts CreationOptions) (gimlet.UserManager, error) {
+	m := &oktaUserManager{
+		clientID:     opts.ClientID,
+		clientSecret: opts.ClientSecret,
+		redirectURI:  opts.RedirectURI,
+		issuer:       opts.Issuer,
+	}
+	return m, nil
+}
+
+func (m *oktaUserManager) GetUserByToken(ctx context.Context, token string) (gimlet.User, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *oktaUserManager) CreateUserToken(user string, password string) (string, error) {
+	return "", errors.New("not implemented")
+}
+
+func (m *oktaUserManager) GetLoginHandler(callbackURL string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Is this able to redirect the user back to the page they were trying to visit once authenticated?